### PR TITLE
Raise error if annotation dict in simple training style has unexpected keys #4074

### PR DESCRIPTION
**Pull request by jenojp**
# Raise error if annotation dict in simple training style has unexpected keys #4074

## Description
Added strict check to see if top level keys in simple annotation dictionary are in list of expected keys and if not, raise ValueError.

### Types of change
Ehancement from issue #4074 

## Checklist
- [X] I have submitted the spaCy Contributor Agreement.
- [X] I ran the tests, and all new and existing tests passed.
- [X] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/language.py [hunk 1]
```diff
@@ -445,6 +445,10 @@ def update(self, docs, golds, drop=0.0, sgd=None, losses=None, component_cfg=Non
             if isinstance(doc, basestring_):
                 doc = self.make_doc(doc)
             if not isinstance(gold, GoldParse):
+                expected_keys = ("words", "tags", "heads", "deps", "entities", "cats")
+                unexpected_keys = [k for k in gold if k not in expected_keys]
+                if unexpected_keys:
+                    raise ValueError(Errors.E151.format(unexpected_keys=unexpected_keys))
```
Comment (svlandeg) [inline spacy/language.py]: Lets add "links" to this list to ensure the upcoming entity linking functionality is already accounted for :-)

File: spacy/errors.py [hunk 2]
```diff
@@ -417,6 +417,10 @@ class Errors(object):
             "component matches the model being loaded.")
     E150 = ("The language of the `nlp` object and the `vocab` should be the same, "
             "but found '{nlp}' and '{vocab}' respectively.")
+    E151 = ("Trying to call nlp.update without required annotation types. "
+            "Expected top-level keys 'words', 'tags', 'heads', 'deps', 'entities' or 'cats'."
+            " Got: {unexpected_keys}.")
+
```
Comment (svlandeg) [inline spacy/errors.py]: Could we somehow make sure to define this list only once? Because also here we'd want to include "links" (cf other comment)

Comment (ines) [reply spacy/errors.py]: Yes, that's a good idea! We can just pass in the `expected_keys` here.

Comment (jenojp) [reply spacy/errors.py]: Good ideas! I put both of these in the latest commit

Comment (svlandeg) [reply spacy/errors.py]: Looks good!

**Comment by honnibal** [comment]
Thanks for your help! Looks good.

**Pull request merged**